src/controllers/HierarchicalFacetSearchController.ts

```diff
@@ -0,0 +1,59 @@
+import { IDynamicHierarchicalFacet } from '../ui/DynamicHierarchicalFacet/IDynamicHierarchicalFacet';
+import { FacetSearchType, IFacetSearchRequest } from '../rest/Facet/FacetSearchRequest';
+import { IFacetSearchResponse } from '../rest/Facet/FacetSearchResponse';
+import { QueryUtils, DateUtils } from '../Core';
+import { FileTypes } from '../ui/Misc/FileTypes';
+
+export class HierarchicalFacetSearchController {
```

Comment:
This class is for the most part a copy of `FacetSearchController`. The differences are mostly in the `search` method.